### ISSUE TITLE
Che 7.10.0 Release

### ISSUE THREAD

**Issue by nickboldt** [eclipse-che/che#16298]
### List of pending issues / PRs

- [ ] *(add blockers here)*

### Release status

In parallel, the following releases can be done:

* che-theia,
* che-machine-exec,
* che-devfile registry, then che-plugin-registry (once che-theia and machine-exec are done)
* che-parent, then che-docs, then che as Release Candidate for QE

Then in series:

* che-parent, then che-docs, then che as Release
* che-operator, then chectl

- [x] che-theia _*(d0e00c5d07ec0c95a17da51a5b744934d7c4d1f8)*_
- [x] che-machine-exec
- [x] che-plugin-registry _(depends on che-theia, che-machine-exec)_
- [x] che-devfile-registry
- [x] che-parent
- [x] che-docs _(depends on parent)_
- [x] che _(depends on docs and parent)_
- [x] che-operator _(depends on all of the above)_
  - [x] Kubernetes community operator PR https://github.com/operator-framework/community-operators/pull/1342
  - [x] OpenShift community operator PR https://github.com/operator-framework/community-operators/pull/1341
- [x] chectl _(depends on all of the above)_
- [x] complete current milestone
  - [x] move incomplete *deferred* issues to backlog
  - [x] move incomplete *WIP* issues to next milestone
  - [x] close completed issues
  - [x] close milestone

| Owner | Process | Script | CI | Artifact(s) |
| --- | --- | --- | --- | --- |
| @azatsarynnyy| [che-theia](https://github.com/eclipse/che-theia/blob/master/RELEASE.md) | [***MANUAL***](https://github.com/eclipse/che-theia/blob/master/RELEASE.md) | [centos](https://ci.centos.org/job/devtools-che-theia-che-release/) | [`quay.io/eclipse/che-theia`](https://quay.io/eclipse/che-theia) |
| @nickboldt| [che-machine-exec](https://github.com/eclipse/che-machine-exec/blob/master/RELEASE.md) | [make-release.sh](https://github.com/eclipse/che-machine-exec/blob/master/make-release.sh) | [centos](https://ci.centos.org/job/devtools-che-machine-exec-release/) | [`quay.io/eclipse/che-machine-exec`](https://quay.io/eclipse/che-machine-exec)| 
| @nickboldt| [che-devfile-registry](https://github.com/eclipse/che-devfile-registry/blob/master/RELEASE.md) | [make-release.sh](https://github.com/eclipse/che-devfile-registry/blob/master/make-release.sh) | [centos](https://ci.centos.org/job/devtools-che-devfile-registry-release/) | [`quay.io/eclipse/che-devfile-registry`](https://quay.io/eclipse/che-devfile-registry)| 
| @nickboldt / @ericwill| [che-plugin-registry](https://github.com/eclipse/che-plugin-registry/blob/master/RELEASE.md) | [make-release.sh](https://github.com/eclipse/che-plugin-registry/blob/master/make-release.sh) | [centos](https://ci.centos.org/job/devtools-che-plugin-registry-release/) | [`quay.io/eclipse/che-plugin-registry`](https://quay.io/eclipse/che-plugin-registry)| 
| @vparfonov / @mkuznyetsov| [che-parent](https://github.com/eclipse/che/blob/master/RELEASE.md) | [***MANUAL***](https://github.com/eclipse/che/blob/master/RELEASE.md) | [centos](https://ci.centos.org/job/devtools-che-parent-che-release/) | [che-parent](https://search.maven.org/search?q=a:che-parent), [che.depmgt](https://search.maven.org/artifact/org.eclipse.che.depmgt/maven-depmgt-pom) |
| @vparfonov / @mkuznyetsov| [che-docs](https://github.com/eclipse/che/blob/master/RELEASE.md) | [***MANUAL***](https://github.com/eclipse/che/blob/master/RELEASE.md) | [centos](https://ci.centos.org/job/devtools-che-docs-che-release/) | [che-docs](https://search.maven.org/search?q=a:che-docs)
| @vparfonov / @mkuznyetsov| [che](https://github.com/eclipse/che/blob/master/RELEASE.md) | [***MANUAL***](https://github.com/eclipse/che/blob/master/RELEASE.md) | [centos](https://ci.centos.org/job/devtools-che-che-release/) | [che.core](https://search.maven.org/search?q=che.core), [`quay.io/eclipse/che-server`](https://quay.io/eclipse/che-server) |
| @tolusha| [che-operator](https://github.com/eclipse/che-operator/blob/master/RELEASE.md) | [make-release.sh](https://github.com/eclipse/che-operator/blob/master/make-release.sh) | ? | [`quay.io/eclipse/che-operator`](https://quay.io/eclipse/che-operator)| 
| @tolusha| [chectl](https://github.com/che-incubator/chectl/blob/master/RELEASE.md) | [make-release.sh](https://github.com/che-incubator/chectl/blob/master/make-release.sh) | [travis](https://travis-ci.org/che-incubator/chectl) | [chectl releases](https://github.com/che-incubator/chectl/releases)

**Comment by nickboldt** [comment]
che-theia, machine-exec, and registries are done.

Waiting on parent, docs, and server to go to RC next. @vparfonov please update this issue when you're in RC and again when RC has passed QE & release is done.

**Comment by vparfonov** [comment]
RC images updated and ready for testing

**Comment by dmytro-ndp** [comment]
Che QE team is preparing Jenkins CI jobs to test Eclipse Che 7.10 https://github.com/eclipse/che/issues/16323

**Comment by dmytro-ndp** [comment]
Current result against Multiuser Eclipse Che 7.10.0.RC without TLS support which was installed using chectl/0.0.20200311-next.938b592 and che-operator:next installer:
1. Java selenium E2E tests passed on minishift 3.11 https://ci.centos.org/view/Devtools/job/devtools-che-rc-multiuser-integration-tests/25/
2. OS OAuth tests passed on minishift 3.11 https://ci.centos.org/view/Devtools/job/devtools-che-rc-ocp-oauth-test/26/
3. Rolling update tests passed on minishift 3.11 https://ci.centos.org/view/Devtools/job/devtools-che-rc-rolling-strategy-test/26/
4. Happy path tests, Git SSH flow tests, Vert.x, Java Maven and Python devfile tests passed on minikube 1.1.1 https://codeready-workspaces-jenkins.rhev-ci-vms.eng.rdu2.redhat.com/view/che-regular-tests/view/pre-release/job/MultiUser-Che-release-check-e2e-tests-against-k8s/40/
 
At the same time installation of Single User Eclipse Che 7.10.0 RC with TLS support using helm installer on minikube had failed https://github.com/eclipse/che/issues/16338
Actually we are investigating the problem.

**Comment by dmytro-ndp** [comment]
@vparfonov: I have found 7.11.0 tags in ["Release preparation" commit](https://github.com/eclipse/che/commit/1eaeb833296b723f1a2671eb1e62a0e954d729a0#diff-2de0b0a873741f8e0d0879a46f8d5266). Shouldn't it be 7.10.0?

**Comment by vparfonov** [comment]
fixed

**Comment by dmytro-ndp** [comment]
@vparfonov: it seems there is improper version of Eclipse Che in quay.io//eclipse/che-server:rc  
SHA256 2e74e658fa4e image - 7.9.1-SNAPSHOT:
![Screenshot from 2020-03-12 19-23-27](https://user-images.githubusercontent.com/1197777/76548404-54934b80-6497-11ea-8a3c-dff82ba59ea4.png)

**Comment by dmytro-ndp** [comment]
@vparfonov: we checked Eclipse Che RC installation twice - there is "7.10.0-SNAPSHOT" label.
![rc-che-local](https://user-images.githubusercontent.com/1197777/76603572-df1a9000-6515-11ea-830b-faee9fd395f6.png)

**Comment by dmytro-ndp** [comment]
An issue [Make embedded OAuth authentication model default for both singe and multiuser che](https://github.com/eclipse/che/issues/15672) has been checked successfully.

**Comment by dmytro-ndp** [comment]
Vert.x, Java Maven and Python devfile tests passed against Single User Eclipse Che 7.10.0.RC with TLS mode being installed using helm on minikube 1.1.1: https://codeready-workspaces-jenkins.rhev-ci-vms.eng.rdu2.redhat.com/view/che-regular-tests/view/pre-release/job/Single-Che-pre-release-check-test-all-devfiles-against-K8S/30/

**Comment by dmytro-ndp** [comment]
Git SSH Flow tests have passed against Single User Eclipse Che 7.10.0.RC with TLS mode being installed using helm on minikube 1.1.1 https://codeready-workspaces-jenkins.rhev-ci-vms.eng.rdu2.redhat.com/job/Single-Che-pre-release-check-Git-SSH-flow-test-against-K8S/2/

**Comment by dmytro-ndp** [comment]
@vparfonov , @nickboldt, @ibuziuk : no blockers found during pre-release testing of Eclipse Che 7.10.0.RC, which included:
- quay.io/eclipse/che-server:rc
- quay.io/eclipse/che-keycloak:rc
- quay.io/eclipse/che-theia:7.10.0
- quay.io/eclipse/che-theia-endpoint-runtime-binary:7.10.0
- quay.io/eclipse/che-devfile-registry:7.10.0
- quay.io/eclipse/che-plugin-registry:7.10.0
- quay.io/eclipse/che-plugin-artifacts-broker:v3.1.1
- [chectl:next](https://github.com/che-incubator/chectl/releases/tag/20200311080209)
- che-operator:nightly
- centos/postgresql-96-centos7:9.6

If there is nothing else to check, you can release provided release candidate.

**Comment by dmytro-ndp** [comment]
I would only suggest to express somewhere in documentation, that Eclipse Che 7.10.0 is now being installed with TLS enabled by default, and that the next workaround is required to install it on minikube with custom helm chart (**-t** chectl option):
```
we need to put https://raw.githubusercontent.com/che-incubator/chectl/master/installers/cert-manager.yml into the che/installers directory
```

**Comment by vparfonov** [comment]
Maven artifacts 7.10.0 ready [link](https://oss.sonatype.org/service/local/repositories/releases/content/org/eclipse/che/assembly-main/7.10.0/assembly-main-7.10.0.zip)
Docker images available [link](https://quay.io/repository/eclipse/che-server?tab=tags)

**Comment by tolusha** [comment]
`che-operator`, `chectl` 7.10.0 have been released.

**Comment by ibuziuk** [comment]
@nickboldt I guess this one can be closed, right?

**Comment by nickboldt** [comment]
Dude. If you see an issue is done, just close it.
